tests/tests.js

```diff
@@ -175,3 +175,12 @@ test.skip('works with very large lengths', function (t) {
 	t.end();
 });
 */
+
+test('returns the correct name when called with toString', function (t) {
+	function nameOf(fn) {
+		return Function.prototype.toString.call(fn).match(/function\s*([^\s]*)\(/)[1];
```

Comment:
fwiw `Function.toString` works as well, and you may want to use `RegExp#test` instead of `String#match` for a bit cleaner of a test.